uwblueprint/paramedics-react
# RTU Stage 2: Push updates to CCP dashboard

[Notion ticket](https://www.notion.so/uwblueprintexecs/RTU-Stage-3-Push-updates-to-CCP-dashboard-d350cbe422d446d5a45114d5b931c962)

[Figma link](https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-(S20%2C-F20)?node-id=1972%3A36028)

Linked to: uwblueprint/paramedics-web#80

## Changes
- Sets up websocket connection with backend
- Real time updates automatically update patient tables on CCP dashboards (rtu updates client cache)
- Adds 5 second highlight for last updated patient (both patients updated by client and changes from RTU)
- refactors the patient details dialog to stop refetching from cache upon each patient row being clicked
- all patient deleted instance are changed to use the DELETE_PATIENT mutation for consistency 

## Screenshots
![ezgif com-gif-maker](https://user-images.githubusercontent.com/38408276/100172890-53744400-2e97-11eb-8c81-2f0677bd06b6.gif)

## Testing
Checkout ct-setup-websockets branch for paramedics-web before testing (uwblueprint/paramedics-web#80)

- Expected behavior: updating information for any existing patient in a CCP should update that patient for other clients on the same event and ccp page. (test only on the main ccp dashboard page and patient details dialog)
- go to a ccp populated with patients. Open several clients on that route (ex. http://localhost:3000/events/3/ccps/1)
- update patient details on one client. save changes and check that the other client has received those updates.
- try repeating the above steps with multiple clients and try adding and deleting patients
- Changes should persist across multiple CCPs of the same event. Try updating patient while one client is on a different CCP, than return to that CCP and check that updates were received
- check that highlighting persists for 5 seconds for each new updated patient

## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
